packages/viewer/src/lib/rendering/matrices.ts

```diff
@@ -155,3 +155,29 @@ export function makeLookAtMatrix(
   ];
   /* eslint-enable prettier/prettier */
 }
+
+export function experimentalMakeTransformationMatrix(
+  normal1: Vector3.Vector3,
+  position1: Vector3.Vector3,
+  normal2: Vector3.Vector3,
+  position2: Vector3.Vector3
+): Matrix4.Matrix4 {
+  const translationDeltaMatrix = Matrix4.makeTranslation(
+    Vector3.subtract(position2, position1)
+  );
+  const angle = Vector3.angleTo(normal2, normal1);
+
+  const axisDirection = Vector3.normalize(Vector3.cross(normal1, normal2));
+
+  const translation = Matrix4.makeTranslation(position1);
+
+  const rotationMatrix = Matrix4.makeRotation(
+    Quaternion.fromAxisAngle(axisDirection, angle)
+  );
+
+  const r = Matrix4.multiply(
+    Matrix4.multiply(translation, rotationMatrix),
+    Matrix4.makeTranslation(Vector3.negate(position1))
+  );
+  return Matrix4.multiply(translationDeltaMatrix, r);
```

Comment:
The translation matrices are a little overly complicated I believe.  `translation` represents `+ position1` while `translationDeltaMatrix` represents `position2 - position1`.  Simplifying, you get:
```
(position2 - position1) + position1
== position2
```

I.e. the two consecutive left-multiplies by translation matrices can be combined into one simpler matrix.  This works because when left-multiplying by translation matrices, the [rotation part of the right matrix has no affect on the translation values of the left matrix](https://github.com/Vertexvis/vector-math-lib-java/blob/b252e536b873ef65eb5046c50dfcf7372cff79c4/src/main/java/com/vertexvis/vectormath/AffineTransform3f.java#L31).